src/mono/wasm/runtime/invoke-cs.ts

```diff
@@ -122,6 +123,119 @@ export function mono_wasm_bind_cs_function(fully_qualified_name: MonoStringRef,
     }
 }
 
+function mono_wasm_bind_cs_function_naot(fully_qualified_name: CharPtr, fully_qualified_name_length: number, signature_hash: number, signature: JSFunctionSignature, is_exception: Int32Ptr): void {
```

Comment:
@SingleAccretion This is the best backward compatible approach I can think of. Do you have any better suggestion?

Reply:
It is a lot of code. Let me see what all of it does.

Reply:
You look at the initial commit in this branch. That's the minimal change

Reply:
Alright, I looked at it and both `mono_wasm_bind_cs_function_naot` and `mono_wasm_get_assembly_exports_naot` look good to me. I would kindly ask to collapse `mono_wasm_bind_js_function` back with `if`s.